### DISCUSSION TITLE
In the getOrCreatePipelineLayout function, why are 62 VkDescriptorSetLayoutBinding allocated in one time? Do the shaders correspond to 62 samplers bindings??

### DISCUSSION THREAD

**Discussion by zy4944796** [google/filament#6932] [General]
I am learning the vulkan backend code in filament,and I have read some strange code that created 62 bindings at once. Why would you do that? Do shaders even use that many samplers? I can't find the corresponding sampler declaration in the shader files.
![Screenshot from 2023-06-30 20-10-49](https://github.com/google/filament/assets/8578753/37ebd022-df51-4dc4-ac8e-440ebf3c9f8b)

**Comment by poweifeng** [comment]
This is just setting up the layout of the pipeline. Because Filament has a variety of materials and also different render pass parameters; it's not efficient to create pipelinelayout on the fly.  So we create (and cache) generic pipeline layouts, but the actual binding of the descriptors is a much smaller set.
